syntax = "proto2";

package mrpc;

message RpcMeta{
    // common part------------
    // message type
    enum Type{
        REQUEST = 0;
        RESPONSE = 1;
    }
    required Type type = 1;

    // message sequence id
    required uint64 sequence_id = 2;
    // -------------common part

    // request part----------------

    // service name and method name
    optional string service = 101;
    optional string method = 102;

    // Todo add timeout
    // optional int64 server_timeout = 103;

    // ----------------request part

    // response part----------------

    // set true if the call is failed
    optional bool failed = 201;
    
    // the error reason if the call is failed
    optional string reason = 202;

    // ----------------response part
};